src/go/rpk/pkg/cli/cloud/login.go

```diff
@@ -237,3 +252,67 @@ func loginProfileFlow(ctx context.Context, fs afero.Fs, y *config.RpkYaml, auth
 	}
 	return msg, y.Write(fs)
 }
+
+func clusterList(ctx context.Context, cl *cloudapi.Client) (vcs []cloudapi.VirtualCluster, cs []cloudapi.Cluster, err error) {
+	g, egCtx := errgroup.WithContext(ctx)
+	g.Go(func() (rerr error) {
+		vcs, rerr = cl.VirtualClusters(egCtx)
+		if rerr != nil {
+			return fmt.Errorf("unable to get the list of virtual clusters: %v", rerr)
+		}
+		return nil
+	})
+	g.Go(func() (rerr error) {
+		cs, rerr = cl.Clusters(egCtx)
+		if rerr != nil {
+			return fmt.Errorf("unable to get the list of clusters: %v", rerr)
+		}
+		return nil
+	})
+	if err := g.Wait(); err != nil {
+		return nil, nil, err
+	}
+	return
+}
+
+type nameAndCluster struct {
+	name string
+	c    *cloudapi.Cluster
+	vc   *cloudapi.VirtualCluster
+}
```

Comment:
This is admittedly persnickety, but it'd be helpful and informative to declare this just before its first use (175 I think?) and provide some comment on the purpose.

Reply:
It's outside of the `loginProfileFlow` scope because it's also used by `clusterNames` function. I'll move it above to L121